### PR TITLE
Convert the lsp "root" to a monorepo root, seed it with an empty core package

### PR DESCRIPTION
**Pull request by awschristou**
This is the first in a series of changes that remakes the `lsp` folder into a monorepo that we can perform experiments on.

I've left all of the existing code in place for this change, but we don't compile it. Upcoming changes will move or delete the existing code.

The monorepo has been created based using guidance from https://earthly.dev/blog/setup-typescript-monorepo/

An empty package `core/aws-lsp-core` has been set up as a supporting core library, so that CI has something to compile.

If/when we decide to start publishing packages, the repo will be updated with a formal namespace. For now it will be set to `@lsp-placeholder`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lsp/package.json [hunk 1]
```diff
@@ -1,64 +1,35 @@
...
-        "chai": "^4.3.7",
-        "create-temp-directory": "^2.2.0",
-        "eslint": "^8.26.0",
+        "@types/node": "^20.2.5",
+        "@typescript-eslint/eslint-plugin": "^5.59.9",
+        "@typescript-eslint/parser": "^5.59.9",
+        "eslint": "^8.42.0",
         "eslint-plugin-unused-imports": "^2.0.0",
         "husky": "^8.0.3",
-        "mocha": "^10.2.0",
```
Comment (nkomonen-amazon) [inline lsp/package.json:54]: Do the tests still run with some of these dependencies being removed?

File: lsp/tsconfig.json [hunk 2]
```diff
@@ -3,12 +3,17 @@
...
         "target": "ES2021",
         "moduleResolution": "node",
-        "lib": ["ES2021"],
-        "outDir": "out",
-        "sourceMap": true,
+        "rootDir": "./src",
+        "outDir": "./out",
+        "declaration": true,
         "skipLibCheck": true,
+        "forceConsistentCasingInFileNames": true,
```
Comment (nkomonen-amazon) [inline lsp/tsconfig.json:10]: Nice addition

**Review by nkomonen-amazon — APPROVED** [review]
LGTM

**Review by justinmk3 — APPROVED** [review]

File: lsp/package.json (hunk 1 above)
Comment (awschristou) [reply lsp/package.json:54]: The tests will not run currently. I will have a follow up change that moves the "core" code around, wires up the existing tests, and sets the CI to run the tests.

**Pull request merged**